src/Illuminate/Foundation/Console/TinkerCommand.php

```diff
@@ -26,11 +35,66 @@ class TinkerCommand extends Command {
 	 */
 	public function fire()
 	{
-		$this->getApplication()->setCatchExceptions(false);
+		$app = $this->getApplication();
+		$app->setCatchExceptions(false);
 
-		$shell = new Shell();
+		$config = new Configuration();
+
+		$pm = $config->getPresenterManager();
+		$pm->addPresenters($this->getPresenters());
+
+		$shell = new Shell($config);
+		$shell->addCommands($this->getCommands());
+		$shell->setIncludes($this->argument('include'));
+		$shell->setScopeVariables([
+			'app' => $app->getLaravel(),
+		]);
 
 		$shell->run();
 	}
 
+	/**
+	 * Get the console command arguments.
+	 *
+	 * @return array
+	 */
+	protected function getArguments()
+	{
+		return [
+			['include', InputArgument::IS_ARRAY, 'Include file(s) before starting tinker'],
+		];
+	}
+
+	/**
+	 * Get artisan commands to pass through to PsySH.
+	 *
+	 * @return array
+	 */
+	protected function getCommands()
+	{
+		$app = $this->getApplication();
+		$commands = [];
+
+		foreach ($app->all() as $name => $command) {
```

Comment:
And one-line Closures, and sometimes one-line `if` and `foreach` blocks, and empty methods, and for some reason sometimes one-line methods. ;)